stout/flags.cc

```diff
@@ -113,6 +121,52 @@ void Parser::Parse(int* argc, const char*** argv) {
     values.emplace(name, value);
   }
 
+  // Parse environment variables if environment_variable_prefix_
+  // contains the specific prefix of variables we want to parse.
+  if (environment_variable_prefix_) {
+    // Run through all environment variables and select those which has
+    // prefix (environment_variable_prefix_) in the name. Then we grab
+    // correct name and the value and store this pair in `values`.
+    // We use global variable `environ` to have access to the environment
+    // variables to be able to select all necessary variables for
+    // parsing according to the included prefix (check class `Parser`
+    // in the `stout/flags.h` which has `environment_variable_prefix_`
+    // variable of the type `std::optional<std::string>`).
+    for (char** env = environ; *env != nullptr; ++env) {
+      if (!absl::StrContains(*env, *environment_variable_prefix_)) {
+        continue;
+      }
+
+      // By default all environment variables are as follows:
+      //    name=value
+      // That's why we need to grab correctly the part with name
+      // and the part with value.
+      const std::vector<std::string> name_value =
+          absl::StrSplit(*env, absl::MaxSplits('=', 1));
+
+      CHECK_EQ(name_value.size(), 2u)
+          << "Expecting all environment variables to have '=' delimiter";
+
+      // Grab the name and the value.
+      absl::string_view name{name_value[0]};
+      std::string value{name_value[1]};
+
+      if (absl::ConsumePrefix(&name, *environment_variable_prefix_)) {
+        // Check the first symbol of the name without prefix.
+        // Remove first character if the symbol is not a let-
+        // ter or a digit.
+        if (!absl::ascii_isalpha(name.at(0))
+            && !absl::ascii_isdigit(name.at(0))) {
+          name.remove_prefix(1);
+        }
```

Comment:
I don't understand this case and don't seem to see a test for it below either, can you elaborate why we need to do this?

Reply:
I've decided to introduce this case because of the following reason:  users can set environment variable like `SOME_PREFIX_NAME`. Then if users will call `IncludeEnvironmentVariablesWithPrefix("SOME_PREFIX_")`
everything will be ok. BUT if they will call `IncludeEnvironmentVariablesWithPrefix("STOUT_PREFIX")` - the result of the parsing will be the following:
`name = '_NAME' `. That's why I check the first character in the name, and if it is not a letter I just remove first character. Actually I think that `!absl::ascii_isdigit(name.at(0))` is redundant cause the names cannot start with digits, so I will remove this check from the #159 line.
Also I will add test to cover this case too.